rayon-core/src/lib.rs

```diff
@@ -144,6 +144,13 @@ impl Configuration {
         if self.num_threads > 0 {
             self.num_threads
         } else {
+            match env::var("RAYON_RS_NUM_THREADS").ok().and_then(|s| usize::from_str(&s).ok()) {
+                Some(x) if x > 0 => return x,
+                Some(x) if x == 0 => return num_cpus::get(),
+                _ => {},
+            }
+
+            // Support for deprecated `RAYON_RS_NUM_CPUS`.
             match env::var("RAYON_RS_NUM_CPUS").ok().and_then(|s| usize::from_str(&s).ok()) {
```

Comment:
I am not sure if a message along the lines of “`RAYON_RS_NUM_CPUS` is deprecated and will be removed in the future; please use `RAYON_RS_NUM_THREADS`” would be appropriate? At first I intended to move this match into a function, but then I realize that the `#[deprecated]` attribute is a compile time feature.